### ISSUE TITLE
Backup completes successfully even though it has errors

### ISSUE THREAD

**Issue by arianitu** [heptio/velero#1423]
**What steps did you take and what happened:**
I'm running into edge cases with GCP and retention-policies, but it is showing a few edge cases that may be useful for Velero to fix.

First one:

1. Backup completes even though an error updating the revision happens:

```
time="2019-05-01T18:24:46Z" level=info msg="Backup completed successfully" backup=velero/testing-ari logSource="pkg/backup/backup.go:288"
time="2019-05-01T18:24:47Z" level=warning msg="Error updating backup store revision" backup=testing-ari bucket=velero-backups error="error updating revision file: rpc error: code = Unknown desc = googleapi: Error 403: Object 'velero-backups/metadata/revision' is subject to bucket's retention policy and cannot be deleted, overwritten or archived until 2019-05-16T10:41:44.818458602-07:00, forbidden" error.file="/go/src/github.com/heptio/velero/pkg/persistence/object_store.go:388" error.function="github.com/heptio/velero/pkg/persistence.(*objectBackupStore).putRevision" logSource="pkg/persistence/object_store.go:206" prefix=
time="2019-05-01T18:24:47Z" level=info msg="Backup completed" backup=velero/testing-ari logSource="pkg/controller/backup_controller.go:492"
```

2. Backup also completes without snapshotting anything if you set your region incorrectly in the yaml file. In my case, I set region instead of snapshotLocation (which is required on GCP, region is for AWS.)

```
time="2019-05-01T18:22:24Z" level=error msg="Error getting volume snapshotter for volume snapshot location" backup=velero/testing-ari error="rpc error: code = Unknown desc = config has invalid keys [region]; valid keys are [snapshotLocation]" error.file="/go/src/github.com/heptio/velero/pkg/cloudprovider/config.go:50" error.function=github.com/heptio/velero/pkg/cloudprovider.validateConfigKeys group=v1 groupResource=persistentvolumeclaims logSource="pkg/backup/item_backupper.go:416" name=pvc-d22d2f57-13c5-11e9-8bea-42010aa2004f namespace=smartapp persistentVolume=pvc-d22d2f57-13c5-11e9-8bea-42010aa2004f volumeSnapshotLocation=default
```

**What did you expect to happen:**
The backup should be set to failed for both cases.

**Environment:**
GCP

- Velero version (use `velero version`): 
Client:
	Version: v1.0.0-alpha.2
	Git commit: dda76b05a8f970ee03bc7fa5bee35bef2f70127f
Server:
	Version: v1.0.0-alpha.2

- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.2", GitCommit:"81753b10df112992bf51bbc2c2f85208aad78335", GitTreeState:"clean", BuildDate:"2018-05-12T04:12:47Z", GoVersion:"go1.9.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.8-gke.6", GitCommit:"394ee507d00f15a63cef577a14026096c310698e", GitTreeState:"clean", BuildDate:"2019-03-30T19:31:43Z", GoVersion:"go1.10.8b4", Compiler:"gc", Platform:"linux/amd64"}

- Kubernetes installer & version:
GKE

- Cloud provider or hardware configuration:
GKE

- OS (e.g. from `/etc/os-release`):

GKE

**Comment by nrb** [comment]
Thanks for the report! This should be made clearer with https://github.com/heptio/velero/pull/1386, which will be include on our beta1 release coming later today.

**Comment by skriss** [comment]
This has been addressed by #1386, as warnings/errors are recorded and cause the backup to be "PartiallyFailed". Closing out!
